### PR TITLE
Rework delete all

### PR DESCRIPTION
**Pull request by VitalikYaroviy**
https://trello.com/c/ZEEu6W4q

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -3,7 +3,11 @@ class PostsController < ApplicationController
   before_action :set_post, only: [:show, :edit, :update, :destroy]
 
   def index
-    @posts = Post.search(params[:search])
+    @post = Post.search(params[:search])
+    @posts = []
+    @post.each do |post|
```
Comment (eugenekorpan) [inline app/controllers/posts_controller.rb]: change to association

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -34,77 +34,87 @@
         </div>
         <div class="col-sm">
           <div class="d-flex justify-content-end">
-            <%= link_to 'Check all', {controller: :posts, action: :select_all}, :remote => true, class:"mr-4 btn btn-success"%>
-            <%= link_to 'Uncheck all', {controller: :posts, action: :uncheck_all}, :remote => true, class:"mr-4 btn btn-success"%>
-            <%= link_to 'Remove all', {controller: :posts, action: :remove_all}, :remote => true, class:"mr-4 btn btn-warning"%>
+            <%= link_to 'Check all', {controller: :posts, action: :select_all}, :remote => true, class: "mr-4 btn btn-success" %>
+            <%= link_to 'Uncheck all', {controller: :posts, action: :uncheck_all}, :remote => true, class: "mr-4 btn btn-success" %>
+            <%= link_to 'Remove all', {controller: :posts, action: :remove_all}, :remote => true, class: "mr-4 btn btn-warning" %>
```
Comment (eugenekorpan) [inline app/views/posts/index.html.erb]: onClick = 'deleteAll()'

File: app/views/posts/remove_all.js.erb [hunk 3]
```diff
@@ -1,5 +1,13 @@
-$(':checkbox').each(function () {
-    if(this.checked) {
-        $(this).closest('.post').detach();
-    }
+$("input.forRemove[type='checkbox']").each(function () {
```
Comment (eugenekorpan) [inline app/views/posts/remove_all.js.erb]: function deleteAll() {

..
}

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -3,8 +3,8 @@ class PostsController < ApplicationController
   before_action :set_post, only: [:show, :edit, :update, :destroy]
 
   def index
-    @posts = Post.search(params[:search])
-    @post = Post.new
+    # @posts = Post.search(params[:search])
```
Comment (eugenekorpan) [inline app/controllers/posts_controller.rb]: don't push commented out code

File: app/assets/javascripts/remove_all.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+function removeAll() {
+  $("input.forRemove[type='checkbox']").each(function () {
+    if (this.checked) {
+      let arr_tr = $(this).closest('.post')
+      let post_id = arr_tr[0].id;
+
+      $.ajax({
```
Comment (eugenekorpan) [inline app/assets/javascripts/remove_all.js]: rework to send only one ajax request for all the selected posts

File: app/assets/javascripts/remove_all.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+function removeAll() {
+  let arr_ids = [];
+  $("input.forRemove[type='checkbox']").each(function () {
+    if (this.checked) {
+      let post = $(this).closest('.post');
+      let post_id = post[0].id;
```
Comment (eugenekorpan) [inline app/assets/javascripts/remove_all.js]: javascript uses CamelCase syntax style

File: app/assets/javascripts/remove_all.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+function removeAll() {
+  let arr_ids = [];
+  $("input.forRemove[type='checkbox']").each(function () {
```
Comment (eugenekorpan) [inline app/assets/javascripts/remove_all.js]: update css selector to find already checked checkboxes

File: app/assets/javascripts/remove_all.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    if (this.checked) {
+      let post = $(this).closest('.post');
+      let post_id = post[0].id;
+      arr_ids.push(post_id);
+      $(this).closest('.post').detach();
```
Comment (eugenekorpan) [inline app/assets/javascripts/remove_all.js]: this should occur on a successful ajax response

File: app/assets/javascripts/remove_all.js (hunk 6 above)
Comment (eugenekorpan) [inline app/assets/javascripts/remove_all.js]: your checkbox already has post id

File: app/controllers/posts_controller.rb [hunk 9]
```diff
@@ -46,6 +46,7 @@ def update
   end
 
   def destroy
+    @post = Post.find(params[:id])
```
Comment (eugenekorpan) [inline app/controllers/posts_controller.rb]: !

File: app/controllers/posts_controller.rb [hunk 10]
```diff
@@ -62,31 +63,8 @@ def set_post
...
-
-  def uncheck_all
-    Post.where(user_id: current_user.id).update_all(statusDelete: false)
-    respond_to do |format|
-      format.js
-      format.html {redirect_to posts_url}
-      format.json {head :no_content}
-    end
+  def destroy_multiple
+    @post = Post.destroy(params[:ids])
```
Comment (eugenekorpan) [inline app/controllers/posts_controller.rb]: !

File: app/assets/javascripts/remove_all.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    arrIds.push(postId);
+    elements.push(this)
+  });
+
+  $.ajax({
+    url: '/destroy_multiple',
+    type: 'DELETE',
+    data: {ids: arrIds}
+  }).then(function () {
+    $(elements).closest('.post').detach();
```
Comment (eugenekorpan) [inline app/assets/javascripts/remove_all.js]: почитай пліз про різницю між detach() i remove()

File: app/assets/javascripts/remove_all.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
+function removeAll() {
+  let arrIds = [];
+  let elements = [];
+  $("input.forRemove[type='checkbox']:checked").each(function () {
+    let post = $(this).closest('.post');
```
Comment (eugenekorpan) [inline app/assets/javascripts/remove_all.js]: цілком ймовірно що хтось колись може додати чи заюзати клас .post для інших елементів
в такому випадку цей код не буде працювати.
Я б надав більше точності для впевненості що воно завжди знаходитиме те що треба, наприклад `(tr.post)`

**Comment by eugenekorpan** [comment]
:+1:
